### PR TITLE
p2pool: 2.7 -> 3.1

### PR DESCRIPTION
**Pull request by nathanruiz**
###### Description of changes

Release notes:
- https://github.com/SChernykh/p2pool/releases/tag/v3.0
- https://github.com/SChernykh/p2pool/releases/tag/v3.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hx507** [comment]
The manual patch disabling LTO is no longer necessary for p2pool v3.1 and onwards, as the original p2pool have disabled it in flake.nix:

https://github.com/SChernykh/p2pool/commit/14a1918fdadf1766cd1ab0c1301919494ac31734

**Comment by nathanruiz** [comment]
@hx507 I've change this to target 3.1, and to use the flag in the upstream repo. Thanks for the update!

**Review by Lord-Valen — APPROVED** [review]
Tested with xmrig and monerod, they communicate as expected.

File: pkgs/applications/misc/p2pool/default.nix [hunk 1]
```diff
@@ -19,20 +19,22 @@ let
...
-    sha256 = "sha256-j3SVwat/LGw/iGcyNn8acR29Ob/WXDKyeCfDTsH+gxA=";
+    sha256 = "sha256-yHxg/9QhaDNlUFzylftsJEk+pJoSoTfA0rJfcolBdTs=";
     fetchSubmodules = true;
   };
 
   nativeBuildInputs = [ cmake pkg-config ];
   buildInputs = [ libuv zeromq libsodium gss hwloc openssl curl ]
     ++ lib.optionals stdenv.isDarwin [ Foundation ];
 
+  cmakeFlags = ["-DWITH_LTO=OFF"];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/p2pool/default.nix:36]: Shouldn't this be removed?

Comment (Lord-Valen) [reply pkgs/applications/misc/p2pool/default.nix:36]: I don't reckon so. The comment above was referring to a `.patch` file which was removed in the force-push from https://github.com/NixOS/nixpkgs/commit/7e3decb417147f4a1d995bd15e375dd04ca0ca3c to https://github.com/NixOS/nixpkgs/commit/d86c897423d8a5416d5a02b66468407edf36cf9e.

**Pull request merged**